Modules/LancetRobotRegistration/src/LancetRobotRegistration.cpp

```diff
@@ -0,0 +1,274 @@
+}
+
+void LancetRobotRegistration::rxm()
+{
+	double templeArray[3] = { -1, 0.0, 0.0 };
+	m_Robot->Rotate(templeArray, Angle);
+}
+
+void LancetRobotRegistration::rym()
+{
+	double templeArray[3] = { 0.0, -1, 0.0 };
+	m_Robot->Rotate(templeArray, Angle);
+}
+
+void LancetRobotRegistration::rzm()
+{
+	double templeArray[3] = { 0.0, 0.0, -1 };
+	m_Robot->Rotate(templeArray, Angle);
+}
+
+int LancetRobotRegistration::captureRobot()
+{
+	if (m_RobotRegistration.PoseCount() < 5) //The first five translations, 
+	{
+		CapturePose(true);
+	}
+	else if (m_RobotRegistration.PoseCount() >= 5 && m_RobotRegistration.PoseCount() < 10)
+	{
+		CapturePose(false);
+	}
+	else
+	{
+		vtkNew<vtkMatrix4x4> robotEndToFlangeMatrix;
+		m_RobotRegistration.GetTCPmatrix(robotEndToFlangeMatrix);
+		robotEndToFlangeMatrix->Invert();
+		std::cout << "Registration RMS: " << m_RobotRegistration.RMS() << std::endl;
+	}
+	return m_RobotRegistration.PoseCount();
+}
+
+void LancetRobotRegistration::CapturePose(bool translationOnly)
+{
+	std::cout << "CapturePose" <<std::endl;
+	//get T_BaseToFlanger
+	vtkSmartPointer<vtkMatrix4x4>T_BaseToFlange = m_Robot->GetBaseToFlange();
+	PrintDataHelper::CoutMatrix("T_BaseToFlange", T_BaseToFlange);
+	//get T_CameraToEnd
+	vtkSmartPointer<vtkMatrix4x4>T_CameraToEnd = m_Camera->GetToolMatrixByName("RobotEndRF");
+	PrintDataHelper::CoutMatrix("T_CameraToEnd", T_CameraToEnd);
+	//get T_BaseRFToCamera
+	vtkSmartPointer<vtkMatrix4x4>T_BaseRFToCamera= m_Camera->GetToolMatrixByName("RobotBaseRF");
+	T_BaseRFToCamera->Invert();
+
+	PrintDataHelper::CoutMatrix("T_BaseRFToCamera", T_BaseRFToCamera);
+	//get T_BaseRFToEnd
+	vtkSmartPointer<vtkMatrix4x4> T_BaseRFToEnd= vtkSmartPointer<vtkMatrix4x4> ::New();
+
+	vtkMatrix4x4::Multiply4x4(T_BaseRFToCamera, T_CameraToEnd, T_BaseRFToEnd);
+
+	////Robotic arm registration
+	m_RobotRegistration.AddPoseWithVtkMatrix(T_BaseToFlange, T_BaseRFToEnd, translationOnly);
+}
+
+void LancetRobotRegistration::waitMove()
+{
+	std::cout << "waitMove" << std::endl;
+	bool bDone = false;
+	while (!bDone)
+	{
+		int nRet = HRIF_IsMotionDone(0, 0, bDone);
+		if (bDone == false)
+		{
+			QThread::msleep(10);
+			QApplication::processEvents();
+			std::cout << "is still moving" << std::endl;
+		}
+		else
+		{
+			int ret = HRIF_GrpStop(0, 0);
+			std::cout << "robot is stop please continue" << std::endl;
+			return;
+		}
+	}
+}
+void LancetRobotRegistration::Sleep(int msec)
+{
+	QTime dieTime = QTime::currentTime().addMSecs(msec);
+
+	while (QTime::currentTime() < dieTime)
+
+		QCoreApplication::processEvents(QEventLoop::AllEvents, 100);
+}
+void LancetRobotRegistration::autoCollection()
+{
+	std::cout << "Starting automatic registration!" << std::endl;
+	int moveCount = 1;
+	m_RobotRegistration.RemoveAllPose();
+	if (isAutoCollectionFlag)
+	{
+		m_Robot->SetTCPToFlange();
+		isAutoCollectionFlag = FALSE;
+	}
+	while (moveCount <= 10)
+	{
+		m_Robot->GoToInitialPos();
+		waitMove();
+		switch (moveCount) {
+		case 1:
```

Comment:
waitMove();
			Sleep(500);
			m_Camera->UpdateData();
			captureRobot();  move out this switch